src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerTestingControllerTest.java

```diff
@@ -44,6 +48,65 @@ public void setup() {
         holdUntilTimestamp = LocalDateTime.parse("2020-03-27T12:56:10.403975").plusDays(10);
     }
 
+    @Test
+    @Ignore("Ignore until AZURE_SERVICE_BUS_FEATURE_TOGGLE is turned on")
+    public void given_post_event_using_test_rest_endpoints_should_create_task() {
+        String caseIdForTask = getCaseId();
+
+        String messageId = randomMessageId();
+        String eventInstanceId = UUID.randomUUID().toString();
+        LocalDateTime timeStamp = LocalDateTime.now();
+        EventInformation eventInformation = buildEventInformation(eventInstanceId, caseIdForTask,
+                                                                  "wa-dlq-user@fake.hmcts.net", timeStamp);
+        EventInformationRequest createRequest = createRequestWithAdditionalMetadata(eventInformation, null);
+
+        postEventToRestEndpoint(messageId, s2sToken, createRequest)
+            .then()
+            .statusCode(HttpStatus.CREATED.value())
+            .assertThat()
+            .body("MessageId", equalTo(messageId))
+            .body("Sequence", notNullValue())
+            .body("CaseId", equalTo(caseIdForTask))
+            .body("EventTimestamp", equalTo(timeStamp.toString()))
+            .body("FromDlq", equalTo(false))
+            .body("State", equalTo(MessageState.NEW.name()))
+            .body("MessageContent", equalTo(asJsonString(createRequest)))
+            .body("Received", notNullValue())
+            .body("DeliveryCount", equalTo(0))
+            .body("RetryCount", equalTo(0))
+
+            .rootPath("MessageProperties")
+            .body("messageProperty1", equalTo("value1"))
+            .body("messageProperty2", equalTo("value2"));
+
+        Response taskFound = findTasksByCaseId(caseIdForTask, 1);
+
+        String taskId = taskFound
+            .then().assertThat()
+            .body("[0].id", CoreMatchers.notNullValue())
+            .extract()
+            .path("[0].id");
+
+        Response response = findTaskDetailsForGivenTaskId(taskId);
+
+        response.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .and().contentType(MediaType.APPLICATION_JSON_VALUE)
+            .body("caseTypeId.value", is("Asylum"))
+            .body("jurisdiction.value", is("IA"))
+            .body("idempotencyKey.value", CoreMatchers.notNullValue())
```

Comment:
should probably check the value look right. E.g contains messageId?

Reply:
see: https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1457304217#WACCDEventHandlingNFRs-IdempotentKeyGeneration

Reply:
Added a check to compare the idempotency key generated in the test